pkg/vulnsrc/osv/types.go

```diff
@@ -1,12 +1,80 @@
 package osv
 
-import "golang.org/x/vuln/osv"
+import (
+	"time"
+)
 
-// source: https://github.com/golang/vuln/blob/9d39a965865fd1d0030df18602433a01f679fd7d/osv/json.go
+type RangeType string
+
+const RangeTypeGit RangeType = "GIT"
+
+type Ecosystem string
+
+// Module identifies the Go module containing the vulnerability.
+// Note that this field is called "package" in the OSV specification.
+//
+// See https://ossf.github.io/osv-schema/#affectedpackage-field.
+type Module struct {
+	Path      string    `json:"name"`
+	Ecosystem Ecosystem `json:"ecosystem"`
+}
+
+type RangeEvent struct {
+	Introduced string `json:"introduced,omitempty"`
+	Fixed      string `json:"fixed,omitempty"`
+}
+
+type Range struct {
+	Type   RangeType    `json:"type"`
+	Events []RangeEvent `json:"events"`
+}
+
+type ReferenceType string
+
+type Reference struct {
+	Type ReferenceType `json:"type"`
+	URL  string        `json:"url"`
+}
+
+type Affected struct {
+	// The affected Go module. Required.
+	// Note that this field is called "package" in the OSV specification.
+	Module            Module            `json:"package"`
+	Ranges            []Range           `json:"ranges,omitempty"`
+	EcosystemSpecific EcosystemSpecific `json:"ecosystem_specific"`
+}
+
+type Package struct {
+	Path    string   `json:"path,omitempty"`
+	GOOS    []string `json:"goos,omitempty"`
+	GOARCH  []string `json:"goarch,omitempty"`
+	Symbols []string `json:"symbols,omitempty"`
+}
+
+type EcosystemSpecific struct {
+	Packages []Package `json:"imports,omitempty"`
+}
+
+// source: https://github.com/golang/vuln/blob/2e18a6705c6b7f63b802491a33e8b78766cb3822/internal/osv/osv.go
 type Entry struct {
```

Comment:
`golang/vuln` moved `Entry` to `internal` folder. We can't import it.
I copied this struct.